packages/twenty-server/src/modules/messaging/message-import-manager/drivers/imap/services/imap-message-locator.service.ts

```diff
@@ -1,100 +1,122 @@
 import { Injectable, Logger } from '@nestjs/common';
 
-import { type ImapFlow } from 'imapflow';
+import { FetchMessageObject, type ImapFlow } from 'imapflow';
 
-import { findSentMailbox } from 'src/modules/messaging/message-import-manager/drivers/imap/utils/find-sent-mailbox.util';
+import { ImapFindSentMailboxService } from 'src/modules/messaging/message-import-manager/drivers/imap/services/imap-find-sent-mailbox.service';
 
 export type MessageLocation = {
   messageId: string;
-  sequence: number;
+  uid: number;
   mailbox: string;
 };
 
 @Injectable()
 export class ImapMessageLocatorService {
   private readonly logger = new Logger(ImapMessageLocatorService.name);
+  private static readonly FETCH_BATCH_SIZE = 500;
```

Comment:
this is a big change, let's be a bit less ambitious and see how it beaves on production fist